### PR TITLE
Added chapter on TPM emulator (jsc#SLE-17527)

### PR DESCRIPTION
**Pull request by tbazant**
### Description

Added a new chapter on software TPM emulator

### Are there any relevant issues/feature requests?

* jsc#SLE-17527

### To which product versions do the changes apply?

The first column is to be filled by the requester, the second column is to be filled by the doc team after the changes have been backported to the maintenance branches.

| Code 15     | Backport Done
| ----------  | ----------
| [x] 15 SP3  | *(no backport necessary)*
| [ ] 15 SP2  | [ ]
| [ ] 15 SP1  | [ ]
| [ ] 15 SP0  | [ ]

| Code 12     | Backport Done
| ----------  | ----------
| [ ] 12 SP5  | [ ]
| [ ] 12 SP4  | [ ]
| [ ] 12 SP3  | [ ]
| [ ] 12 SP2  | [ ]

### CODE REVIEW

File: xml/vt_qemu_tpm.xml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+ <info>
+  <dm:docmanager xmlns:dm="urn:x-suse:ns:docmanager">
+   <dm:bugtracker></dm:bugtracker>
+  </dm:docmanager>
+ </info>
+ <sect1 xml:id="tpm-introduction">
+  <title>Introduction</title>
+  <para>
+   Trusted Platform Module (TPM) is a cryptoprocessor that secures hardware by using cryptographic keys.
+   For developers who use TMP to develop security features, a software TPM emulator is a convenient solution.
```
Comment (lproven) [inline xml/vt_qemu_tpm.xml:18]:
```suggestion
   For developers who use the TPM to develop security features, a software TPM emulator is a convenient solution.
```

File: xml/vt_qemu_tpm.xml [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+ <title>Software TPM emulator</title>
+ <info>
+  <dm:docmanager xmlns:dm="urn:x-suse:ns:docmanager">
+   <dm:bugtracker></dm:bugtracker>
+  </dm:docmanager>
+ </info>
+ <sect1 xml:id="tpm-introduction">
+  <title>Introduction</title>
+  <para>
+   Trusted Platform Module (TPM) is a cryptoprocessor that secures hardware by using cryptographic keys.
```
Comment (lproven) [inline xml/vt_qemu_tpm.xml:17]:
```suggestion
   The Trusted Platform Module (TPM) is a cryptoprocessor that secures hardware using cryptographic keys.
```

File: xml/vt_qemu_tpm.xml [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+   For developers who use TMP to develop security features, a software TPM emulator is a convenient solution.
+   Compared to a hardware TPM device, the emulator has no limit for the number of guests that can access it.
+   Also, it is simple to switch between versions TPM 1.2 and TPM 2.0.
```
Comment (lproven) [inline xml/vt_qemu_tpm.xml:20]:
```suggestion
   Also, it is simple to switch between TPM versions 1.2 and 2.0.
```

File: xml/vt_qemu_tpm.xml [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+ </sect1>
+ <sect1 xml:id="tpm-installation">
+  <title>Installation</title>
+  <para>
+   To use the software TPM emulator, install the <package>swtpm</package> package:
+  </para>
+<screen>&prompt.sudo;zypper install swtpm</screen>
+ </sect1>
+ <sect1 xml:id="tpm-qemu">
+  <title>Using swtpm with &qemu;</title>
```
Comment (lproven) [inline xml/vt_qemu_tpm.xml:39]: Should we use a [command] tag around this? I see from the next paragraph that it **is** a command.

File: xml/vt_qemu_tpm.xml [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+   <para>
+    swtpm provides three types of interface: socket, chardev, and cuse.
```
Comment (lproven) [inline xml/vt_qemu_tpm.xml:41]: Would [literal] tags around these make it clearer?

File: xml/vt_qemu_tpm.xml [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+  </para>
+<screen>
+&lt;devices>
+ &lt;tpm model='tpm-tis'>
+  &lt;backend type='emulator' version='2.0'/>
+ &lt;/tpm>
+&lt;/devices>
+</screen>
+  <para>
+   &libvirt; will start swtpm for the guest automatically, you do not need to start it manually in advance.
```
Comment (lproven) [inline xml/vt_qemu_tpm.xml:120]:
```suggestion
   &libvirt; will start swtpm for the guest automatically; you do not need to start it manually in advance.
```

File: xml/vt_qemu_tpm.xml [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+   The corresponding permall file will be created in <filename>/var/lib/libvirt/swtpm/<replaceable>VM_UUID</replaceable></filename>.
```
Comment (lproven) [inline xml/vt_qemu_tpm.xml:121]:
```suggestion
   The corresponding <filename>permall</filename> file will be created in <filename>/var/lib/libvirt/swtpm/<replaceable>VM_UUID</replaceable></filename>.
```

File: xml/vt_qemu_tpm.xml [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     Configuring a specific virtualization environment on &productname; is described in <xref linkend="cha-vt-installation"/>.
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     Details about <package>swtpm</package> usage are in its manual page <command>man 8 swtpm</command>.
```
Comment (lproven) [inline xml/vt_qemu_tpm.xml:146]:
```suggestion
     Details about <package>swtpm</package> usage are in its manual page: type <command>man 8 swtpm</command>.
```

File: xml/vt_qemu_tpm.xml [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+   If the guest uses the OVMF firmware, it will measure components with TPM.
+   You can find the event log in <filename>/sys/kernel/security/tpm0/binary_bios_measurements</filename>.
+  </para>
+ </sect1>
+ <sect1 xml:id="qemu-tpm-resources">
+  <title>Resources</title>
+  <itemizedlist>
+   <listitem>
+    <para>
+     Thorough insight into TPM is at the wiki page <link xlink:href="https://en.wikipedia.org/wiki/Trusted_Platform_Module"/>.
```
Comment (lproven) [inline xml/vt_qemu_tpm.xml:136]:
```suggestion
     Wikipedia offers a thorough description of the TPM at the page <link xlink:href="https://en.wikipedia.org/wiki/Trusted_Platform_Module"/>.
```

Comment (lproven) [reply xml/vt_qemu_tpm.xml:136]: "Wiki" is a generic term and I think we should say _which_ wiki.

File: xml/vt_qemu_tpm.xml [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+</screen>
+  <para>
+   &libvirt; will start swtpm for the guest automatically, you do not need to start it manually in advance.
+   The corresponding permall file will be created in <filename>/var/lib/libvirt/swtpm/<replaceable>VM_UUID</replaceable></filename>.
+  </para>
+ </sect1>
+ <sect1 xml:id="tpm-ovmf">
+  <title>TPM measurement with OVMF firmware</title>
+  <para>
+   If the guest uses the OVMF firmware, it will measure components with TPM.
```
Comment (lproven) [inline xml/vt_qemu_tpm.xml:127]: I did not know this abbreviation, so perhaps other readers may not. It stands for Open Virtual Machine Firmware: http://www.linux-kvm.org/downloads/lersek/ovmf-whitepaper-c770f8c.txt

It is an implentation of UEFI for VMs, as opposed to the legacy BIOS. Should we mention this?

**Review by lproven — CHANGES_REQUESTED** [review]

**Review by jfehlig — COMMENTED** [review]
I'm not a TPM expert, but the libvirt and qemu usage LGTM.

File: xml/vt_qemu_tpm.xml (hunk 4 above)
Comment (tbazant) [inline xml/vt_qemu_tpm.xml:39]:
```suggestion
  <title>Using <command>swtpm</command> with &qemu;</title>
```

File: xml/vt_qemu_tpm.xml (hunk 5 above)
Comment (tbazant) [inline xml/vt_qemu_tpm.xml:41]:
```suggestion
    <command>swtpm</command> provides three types of interface: <literal>socket</literal>, <literal>chardev</literal>, and <literal>cuse</literal>.
```

File: xml/vt_qemu_tpm.xml (hunk 8 above)
Comment (tbazant) [reply xml/vt_qemu_tpm.xml:146]: no, brackets are better

Comment (tbazant) [inline xml/vt_qemu_tpm.xml:146]:
```suggestion
     Details about <package>swtpm</package> usage are in its manual page (<command>man 8 swtpm</command>).
```

File: xml/vt_qemu_tpm.xml (hunk 10 above)
Comment (tbazant) [inline xml/vt_qemu_tpm.xml:127]:
```suggestion
   If the guest uses the Open Virtual Machine Firmware (OVMF), it will measure components with TPM.
```

**Review by tbazant — COMMENTED** [review]
Few suggestions

**Pull request merged**